src/Quartz/Impl/JobDetailImpl.cs

```diff
@@ -212,7 +276,7 @@ public JobKey Key
 
                 return key;
             }
-            set
+            internal set
```

Comment:
This one cannot be private since we use it in `ServiceCollectionExtensions.ScheduleJob<T>(this ServiceCollectionQuartzConfigurator options, Action<ITriggerConfigurator> trigger, Action<IJobConfigurator>? job = null)`.

I wonder if:
* we shouldn't throw (in `JobBuilder.Build()`if you attempt to build a **IJobDetail** without having assign a key?
* if makes sense to assign a job key that uses the name and group of the trigger in `ServiceCollectionExtensions.ScheduleJob<T>(this ServiceCollectionQuartzConfigurator options, Action<ITriggerConfigurator> trigger, Action<IJobConfigurator>? job = null)`

Reply:
Generally having job/trigger with just a name is all fine, less friction, but defaulting to trigger's name is problematic - job can have N triggers.